server/src/utils.ts

```diff
@@ -0,0 +1,75 @@
+import * as lsp from 'vscode-languageserver';
+import {URI} from 'vscode-uri';
+
+enum Scheme {
+  File = 'file',
+}
+;
+
+/**
+ * Extract the file path from the specified `uri`.
+ * @param uri
+ */
+export function uriToFilePath(uri: string): string {
```

Comment:
Will you get rid of this method when you fix the conversation above?

Reply:
? this method is still needed.
Perhaps it's less confusing if I rename it to `uriToFsPath` ?

Reply:
Can't you just use the function from https://github.com/angular/vscode-ng-language-service/blob/master/server/src/documents.ts#L109-L111? I'm probably missing something, sorry 😅

Reply:
ahh ... sorry! I should have clarified, in a follow-up PR, the following files will be deleted.
1. `documents.ts`
2. `editorServices.ts`
3. `errors.ts`

and `server.ts` would be totally rewritten. The files in this PR is for the new `server.ts`.

We should never implement custom logic for URI parsing, instead should always call methods in `vscode-uri`.

Reply:
Cool, thanks for the clarification.